src/octoprint/printer/standard.py

```diff
@@ -742,6 +743,17 @@ def get_file_position(self):
 
         return self._comm.getFilePosition()
 
+    def addMarking(self, type):
+        self._markings.append(
+            {
+                "type": type,
+                "time": int(time.time() * 1000),
+            }
+        )
```

Comment:
This needs some way to age out old markings. Imagine the printer is online for a month or something (there's nothing that would speak against that), and being used. The memory would slowly get filled with ancient graph markings that aren't even displayed anymore but will still be kept and also sent to the printer on every single state update. That's bad.

Limit this to the temperature cutoff time (`settings().getInt(["temperature", "cutoff"]) * 60`, see `self._temps`), ideally create another `InvariantContainer` like `TemperatureHistory` (or possibly just reuse that by instancing it again, after renaming it to something like `TimedHistory`).

Also, stick to `time.time()` timestamps on the backend, translate to JS-site timestamps on the frontend.